### PR TITLE
[SYCL][L0] Rewrite interop test with new API

### PR DESCRIPTION
**Pull request by smaslov-intel**
Signed-off-by: Sergey V Maslov <sergey.v.maslov@intel.com>

### CODE REVIEW

File: SYCL/Plugin/interop-level-zero.cpp [hunk 1]
```diff
@@ -7,22 +7,25 @@
 #include <CL/sycl.hpp>
 // clang-format off
 #include <level_zero/ze_api.h>
-#include <CL/sycl/backend/level_zero.hpp>
+#include <sycl/ext/oneapi/backend/level_zero.hpp>
```
Comment (vladimirlaz) [inline SYCL/Plugin/interop-level-zero.cpp:10]: We state Level Zero like an extension.
Shouldn't we add corresponding macro SYCL_EXT_ONEAPI_BACKEND_LEVEL_ZERO in sycl/include/CL/sycl/feature_test.hpp like all other extensions and enable the test only when extension is available?

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

File: SYCL/Plugin/interop-level-zero.cpp (hunk 1 above)
Comment (smaslov-intel) [reply SYCL/Plugin/interop-level-zero.cpp:10]: The macro is there already https://github.com/intel/llvm/blob/e66216673f599d85c056bc3a9dd7ddfd5544ff3d/sycl/include/CL/sycl/feature_test.hpp#L30

Comment (smaslov-intel) [reply SYCL/Plugin/interop-level-zero.cpp:10]: I don't think we have to test it in this test, since we know or implementation supports it.

Comment (vladimirlaz) [reply SYCL/Plugin/interop-level-zero.cpp:10]: But not all implementations and build versions may support it (e.g. HIP or CUDA builds may miss this extension one day).
I think it is good practice to wrap extension test with the check.

Comment (smaslov-intel) [reply SYCL/Plugin/interop-level-zero.cpp:10]: But the test already REQUIRES: level_zero

Comment (smaslov-intel) [reply SYCL/Plugin/interop-level-zero.cpp:10]: Anyways, I have add the check for SYCL_EXT_ONEAPI_BACKEND_LEVEL_ZERO

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**